API.md

```diff
@@ -39,6 +42,30 @@ See also [`reject`](#reject), [`resolve`](#resolve).
 all([ Promise.resolve('a') ]) //=> Promise ['a']
 ```
 
+### assign
+
+```haskell
+// assign : String -> ({ k: v } -> a) -> { k: v }
+```
+
+Accepts a property, function and object.  Sets the property on the object to the result of the function.
+
+```js
+assign('foo', {}, always('bar')) //=> { foo: 'bar' }
```

Comment:
IMHO, the name `assign` is a confusing choice. Implies behavior similar to `Object.assign`.

Reply:
yeah, I'm a little iffy on these names, same with `put`.  I spent some time on thesaurus.com and I left :person_shrugging:.  Any ideas?

Reply:
`put` could maybe be: `setProp`.

Reply:
![image](https://user-images.githubusercontent.com/46536/35572658-18012d36-05a3-11e8-902a-4227eded2132.png)

Reply:
![image](https://user-images.githubusercontent.com/46536/35572720-37ae23fa-05a3-11e8-90d7-d1f01af41c2f.png)

Reply:
I vote for `transply` to connote the transformation (function) and the application (or assignment), but ensconce or bestow would be close seconds ;)

Reply:
I really like `plop` as well.

Reply:
This signature doesn't quite add up with the example to me. This signature says

```haskell
assign : String -> ({ k: v } -> a) -> { k: v }
```

which means I take _two_ parameters
* a `String`
* a `Function` that accepts an Object and returns a value

and then returns an Object.

The definition says in text that is takes a `property`, a `function` and an `object`, but the example is `assign('foo', {}, always('bar'))`, which is a `property`, an `object` and a `function`. It seems a number of things might not be right here.

This is echoed below, but it seems like this is trying to accomplish [ramda's assoc](http://ramdajs.com/docs/#assoc) but changing the argument order so that the data structure is _not_ passed last. What would be a good example of a use case for this?